### PR TITLE
Asset Database Future version check can now be downgraded from an error, via a command line flag + set this flag on Jenkins.

### PR DESCRIPTION
**Pull request by AMZN-stankowi**
If the command line flag is not set, the Asset Processor will emit an error if the Asset Database on disk is a future version, and then shut down, to preserve the database.

If the command flag is set, the Asset Processor will instead print a message that this event occured, and then create a new Asset Database.

Integration from Development.

Signed-off-by: AMZN-stankowi <4838196+AMZN-stankowi@users.noreply.github.com>

## What does this PR do?

Fixes the issue where branch Jenkins jobs would fail due to future asset database versions, by having the job create a new database in that situation.

## How was this PR tested?

Tested locally by making asset databases with future versions and running AP.

### CODE REVIEW

**Comment by AMZN-stankowi** [comment]
I'm submitting this to the exception process right now, it hasn't been approved yet.

**Review by amzn-changml — APPROVED** [review]
LGTM. Approving exception on behalf of SIG-Build

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp
Comment (AMZN-daimini) [inline Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp:1794]: nit: prefer `[[ maybe_unused ]]`

**Review by AMZN-daimini — APPROVED** [review]

**Comment by monroegm** [comment]
I approve on behalf of SIG-Content

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp
Comment (AMZN-stankowi) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp:1794]: I thought we used [[maybe_unused]] when it was unused in some configurations, like a variable that's used in Debug and unused in Profile, and commented out the parameter when it's entirely unused?

I'll switch it for now.

**Comment by AMZN-stankowi** [comment]
I don't want to commit this and vanish for the weekend, so I'll start Automated Review first thing Monday morning and get this in then.

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp
Comment (lumberyard-employee-dm) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp:1794]: If the parameter is complete unused then it should be just be deleted.

**Review by brianherrera — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp
Comment (AMZN-stankowi) [reply Code/Framework/AzToolsFramework/AzToolsFramework/AssetDatabase/AssetDatabaseConnection.cpp]: It's a virtual function and it's used elsewhere

**Comment by tonybalandiuk** [comment]
**Exception Tracking**
[X] 1. (sig-content, @monroegm) Owning SIG requests the exception, by a Sig Representative (Chair/co-chair) adding the PR to the exception requests queue board.
[ ] 2. Owning SIG comments on the PR explaining impact if this code does not go into stabilization.
[X] 3. (sig-build, @amzn-changml ) Unrelated SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
[ ] 4. UX SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
[ ] 5. Testing SIG representative (chair/cochair) approves the exception, adding a comment indicating approval and the SIG they represent.
[ ] (Must occur at some point before SIG-Release approval) Code has been approved for merging by maintainers
[ ] 6. SIG-Release Release Manager or co-Release Manager approves the exception, adding a comment indicating approval and the SIG they represent.
[ ] 7. If all approvals are accepted, then the exception PR may be merged
[ ] 8. Merged PR is verified successful via testing. If Merged PR verification testing failed, consult reps from step 3-5 and discussion occurs on fixing the failure vs backing out change
[ ] 9. Exception is complete

**Comment by tonybalandiuk** [comment]
If this is not submitted, it will be slower to get other stabilization fixes in.

**Comment by AMZN-Dk** [comment]
Approved from SIG testing, the validation work for this is fairly straight forward.

**Comment by rainbj** [comment]
Talking with @AMZN-stankowi offline to verify the new workflow doesn't add any new visual UX workflows.

**Comment by rainbj** [comment]
After speaking with @AMZN-stankowi it appears the changes are in alignment with UX guidelines. Thanks! Sig-UI-UX approved.

**Comment by tonybalandiuk** [comment]
Approved by release manager, on behalf of sig-release.

**Pull request merged**
